src/pg_shard.c

```diff
@@ -2104,50 +2111,8 @@ PgShardProcessUtility(Node *parsetree, const char *queryString,
 	}
 	else if (statementType == T_CopyStmt)
 	{
-		CopyStmt *copyStatement = (CopyStmt *) parsetree;
-		RangeVar *relation = copyStatement->relation;
-		Node *rawQuery = copyObject(copyStatement->query);
-
-		if (relation != NULL)
-		{
-			bool failOK = true;
-			Oid tableId = RangeVarGetRelid(relation, NoLock, failOK);
-			bool isDistributedTable = false;
-
-			Assert(rawQuery == NULL);
-
-			isDistributedTable = IsDistributedTable(tableId);
-			if (isDistributedTable)
-			{
-				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```

Comment:
I meant that you could just make a 1 line change to the existing code by replacing this error. PgShardCopy could just assume that the table is distributed and relation != NULL. That would avoid another layer of nesting in PgShardCopy. Moreover, it keeps all the IsDistributedTable logic in this file.